crates/burn-dataset/src/vision/image_folder.rs

```diff
@@ -611,4 +720,128 @@ mod tests {
             Annotation::MultiLabel(vec![0, 2])
         );
     }
+
+    #[test]
+    pub fn segmask_image_path_to_vec_pixel_depth() {
+        let root = Path::new(SEGMASK_ROOT);
+        // test checkerboard mask
+        const TEST_CHECKERBOARD_MASK_PATTERN: [u8; 64 * 3] = [
+            1, 1, 1, 2, 2, 2, 1, 1, 1, 2, 2, 2, 1, 1, 1, 2, 2, 2, 1, 1, 1, 2, 2, 2, 2, 2, 2, 1, 1,
+            1, 2, 2, 2, 1, 1, 1, 2, 2, 2, 1, 1, 1, 2, 2, 2, 1, 1, 1, 1, 1, 1, 2, 2, 2, 1, 1, 1, 2,
+            2, 2, 1, 1, 1, 2, 2, 2, 1, 1, 1, 2, 2, 2, 2, 2, 2, 1, 1, 1, 2, 2, 2, 1, 1, 1, 2, 2, 2,
+            1, 1, 1, 2, 2, 2, 1, 1, 1, 1, 1, 1, 2, 2, 2, 1, 1, 1, 2, 2, 2, 1, 1, 1, 2, 2, 2, 1, 1,
+            1, 2, 2, 2, 2, 2, 2, 1, 1, 1, 2, 2, 2, 1, 1, 1, 2, 2, 2, 1, 1, 1, 2, 2, 2, 1, 1, 1, 1,
+            1, 1, 2, 2, 2, 1, 1, 1, 2, 2, 2, 1, 1, 1, 2, 2, 2, 1, 1, 1, 2, 2, 2, 2, 2, 2, 1, 1, 1,
+            2, 2, 2, 1, 1, 1, 2, 2, 2, 1, 1, 1, 2, 2, 2, 1, 1, 1,
+        ];
+        assert_eq!(
+            TEST_CHECKERBOARD_MASK_PATTERN
+                .iter()
+                .map(|&x| PixelDepth::U8(x))
+                .collect::<Vec<PixelDepth>>(),
+            image_path_to_vec_pixel_depth(&root.join("annotations").join("mask_checkerboard.png")),
+        );
+
+        // checkerboard image
+        // TODO: investigate why the channels appear to be reversed, i.e (blue, green, red) rather than (red, green, blue)
```

Comment:
Is this still the case? Anyway the segmentation masks should be a 2D image `[H, W]` (single channel) so the test cases might be reworked with the other changes.